yt/utilities/lib/interpolators.pyx

```diff
@@ -163,3 +163,115 @@ def ghost_zone_interpolate(int rf,
                 opos[2] += ods[2]
             opos[1] += ods[1]
         opos[0] += ods[0]
+
+@cython.cdivision(True)
+@cython.wraparound(False)
+@cython.boundscheck(False)
+def replace_nonperiodic_with_extrap(np.ndarray[np.float64_t, ndim=3] field,
+                                    np.ndarray[np.int_t,     ndim=1] ncell_lo,
+                                    np.ndarray[np.int_t,     ndim=1] ncell_hi):
+    """
+    The purpose of this function is to extrapolate data from the interior of
+    the buffer region to replace the given data in the buffer region using a
```

Comment:
this sentence isn't clear to me. In particular, I don't understand what you mean by 

> the given data in the buffer region